### PR TITLE
Add WildFly 16.0.0.Final

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

File: 16.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# This image provides a base for building and running WildFly applications.
+# It builds using maven and runs the resulting artifacts on WildFly 16.0.0 Final
+
+FROM centos/s2i-base-centos7
+
+MAINTAINER Ben Parees <bparees@redhat.com>
```
Comment (adambkaplan) [inline 16.0/Dockerfile]: Please update maintainer to Adam Kaplan <adam.kaplan@redhat.com>

Note that the `MAINTAINER` instruction is deprecated - https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

File: 16.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    MAVEN_VERSION=3.5.4
+
+LABEL io.k8s.description="Platform for building and running JEE applications on WildFly 16.0.0.Final" \
+      io.k8s.display-name="WildFly 16.0.0.Final" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.tags="builder,wildfly,wildfly16" \
+      io.openshift.s2i.assemble-input-files="/wildfly/standalone/deployments;/wildfly/standalone/configuration;/wildfly/provided_modules" \
+      io.openshift.s2i.destination="/opt/s2i/destination" \
+      com.redhat.deployments-dir="/wildfly/standalone/deployments" \
+      maintainer="Ben Parees <bparees@redhat.com>"
```
Comment (adambkaplan) [inline 16.0/Dockerfile]: Update maintainer label to "Adam Kaplan \<adam.kaplan@redhat.com\>"

**Comment by bparees** [comment]
@adambkaplan let me know when you're happy w/ this one and i'll manually merge it over the travis failure.

File: 16.0/s2i/bin/usage [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+cat <<EOF
+This is an S2I WildFly v16.0.0 Final centos base image responsible for consuming a JEE project
+and deploying it onto WildFly application server.
+To use it, install S2I: https://github.com/openshift/source-to-image.
+
+Sample invocation:
+
+s2i build git://github.com/bparees/openshift-jee-sample openshift/wildfly-160-centos wildflytest
```
Comment (adambkaplan) [inline 16.0/s2i/bin/usage]: /s/bparees/openshift/

File: 16.0/test/test-app/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+openshift-jee-sample
+====================
+
+A sample app to be deployed on openshift environments
```
Comment (adambkaplan) [inline 16.0/test/test-app/README.md]: /s/openshift/OpenShift/

**Comment by jamesnetherton** [comment]
Pushed 29fc38e2eb7a838d7b44bbfffafe4287ef00b7b6 which addresses those few nits.

File: 16.0/test/test-app/src/main/java/Test2.bak [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// a dummy file to force an actual javac in the maven build
+public class Test2 {
```
Comment (adambkaplan) [inline 16.0/test/test-app/src/main/java/Test2.bak]: Was this intended to be added?

Comment (jamesnetherton) [reply 16.0/test/test-app/src/main/java/Test2.bak]: No. I lazily cloned the WF 15 and modified bits for WF 16. That file is present elsewhere so I've removed it from the entire codebase in 9a9b38ba0e37848ca4bcc030d87f5c59a190e842.

**Pull request merged**